### DISCUSSION TITLE
Simple way to change from number line to number page when searching through PDF file

### DISCUSSION THREAD

**Discussion by tapyu** [BurntSushi/ripgrep#2224] [General] [answered]
I wonder if it is possible to change the meaningless number line of a matching from a PDF file to the actual page number where that matching occurred in the PDF. Using the preprocessor showed on the [`GUIDE.md`](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#preprocessor):

```
#!/bin/sh

case "$1" in
*.pdf)
  # The -s flag ensures that the file is non-empty.
  if [ -s "$1" ]; then
    exec pdftotext - -
  else
    exec cat
  fi
  ;;
*)
  exec cat
  ;;
esac
```
`ripgrep` search though a text file and prints the matching out along with the number line, which is meaningless. My goal is to make `ripgrep` perform like [`pdfgrep`](https://gitlab.com/pdfgrep/pdfgrep) regarding number line printing.

**Comment by BurntSushi** [comment, marked as answer]
There's no easy way that I know of. ripgrep is primarily a plain text search tool with some limited facilities for searching other kinds of files. It will never give a better user experience than a purpose driven tool.

With that said, maybe check out ripgrep-all. It might do what you want out of the box.
